sdk/cosmos/azure_data_cosmos/src/clients/container_client.rs

```diff
@@ -142,13 +340,125 @@ impl ContainerClientMethods for ContainerClient {
+        req.insert_header(
+            constants::PARTITION_KEY,
+            HeaderValue::from_cow(partition_key.into().into_header_value()?),
+        );
+        req.set_json(&item)?;
+        self.pipeline
+            .send(Context::new(), &mut req, ResourceType::Items)
+            .await
+    }
+
+    async fn replace_item<T: Serialize>(
+        &self,
+        partition_key: impl Into<PartitionKey>,
+        item_id: impl AsRef<str>,
+        item: T,
+
+        #[allow(unused_variables)]
+        // REASON: This is a documented public API so prefixing with '_' is undesirable.
+        options: Option<ItemOptions>,
+    ) -> azure_core::Result<azure_core::Response<Item<T>>> {
+        let url = self
+            .container_url
+            .with_path_segments(["docs", item_id.as_ref()]);
+        let mut req = Request::new(url, azure_core::Method::Put);
+        req.insert_header(
+            constants::PARTITION_KEY,
+            HeaderValue::from_cow(partition_key.into().into_header_value()?),
+        );
+        req.set_json(&item)?;
+        self.pipeline
+            .send(Context::new(), &mut req, ResourceType::Items)
+            .await
+    }
+
+    async fn upsert_item<T: Serialize>(
+        &self,
+        partition_key: impl Into<PartitionKey>,
+        item: T,
+
+        #[allow(unused_variables)]
+        // REASON: This is a documented public API so prefixing with '_' is undesirable.
+        options: Option<ItemOptions>,
+    ) -> azure_core::Result<azure_core::Response<Item<T>>> {
+        let url = self.container_url.with_path_segments(["docs"]);
+        let mut req = Request::new(url, azure_core::Method::Post);
+        req.insert_header(constants::IS_UPSERT, "true");
+        req.insert_header(
+            constants::PARTITION_KEY,
+            HeaderValue::from_cow(partition_key.into().into_header_value()?),
+        );
+        req.set_json(&item)?;
+        self.pipeline
+            .send(Context::new(), &mut req, ResourceType::Items)
+            .await
+    }
+
+    async fn read_item<T: DeserializeOwned>(
+        &self,
+        partition_key: impl Into<PartitionKey>,
+        item_id: impl AsRef<str>,
+
+        #[allow(unused_variables)]
+        // REASON: This is a documented public API so prefixing with '_' is undesirable.
+        options: Option<ItemOptions>,
+    ) -> azure_core::Result<azure_core::Response<Item<T>>> {
+        let url = self
+            .container_url
+            .with_path_segments(["docs", item_id.as_ref()]);
+        let mut req = Request::new(url, azure_core::Method::Get);
+        req.insert_header(
+            constants::PARTITION_KEY,
+            HeaderValue::from_cow(partition_key.into().into_header_value()?),
+        );
+        self.pipeline
+            .send(Context::new(), &mut req, ResourceType::Items)
+            .await
+    }
+
+    #[allow(async_fn_in_trait)] // REASON: See https://github.com/Azure/azure-sdk-for-rust/issues/1796 for detailed justification
```

Comment:
This was copied from the trait and can be deleted, right?

Reply:
Correct!